_source/_posts/2024-04-30-spring-api-fga.md

```diff
@@ -0,0 +1,901 @@
+---
+layout: blog_post
+title: "Adding Fine-Grained Authorization (FGA) to a Spring Boot API with OpenFGA"
+author: jimena-garbarino
+by: contractor
+communities: [security,java]
+description: "How to add Fine-Grained Authorization (FGA) to a Spring Boot API using the OpenFGA Spring Boot starter"
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+
+Fine-grained authorization (FGA) refers to the capability of granting individual users permission to perform particular actions on specific resources. Effective FGA systems enable the management of permissions for a large number of objects and users. These permissions can undergo frequent changes as the system dynamically adds objects and adjusts access permissions for its users.
+
+OpenFGA is an open-source authorization system designed by Okta for developers, and adopted by the Cloud Native Computing Foundation (CNCF). It offers scalability and flexibility for the implementation of RBAC and ABAC authorization models, moving authorization logic outside application code, making it simpler to evolve authorization policies as complexity grows. In this guide, you will learn how to secure a Spring Boot document API with Auth0 and integrate Fine-Grained Authorization (FGA) into the document operations with OpenFGA.
+
+{% img blog/spring-api-fga/logos.png alt:"Spring Boot plus Auth0 by Okta plus OpenFGA" width:"600" %}{: .center-image }
+
+> **This tutorial was created with the following tools and services**:
+> - [Java OpenJDK 21](https://jdk.java.net/java-se-ri/21)
+> - [Auth0 account](https://auth0.com/signup)
+> - [Auth0 CLI 1.4.0](https://github.com/auth0/auth0-cli#installation)
+> - [Docker 24.0.7](https://docs.docker.com/desktop/)
+> - [OpenFGA Server 1.5.1](https://hub.docker.com/r/openfga/openfga)
+> - [FGA CLI v0.2.7](https://openfga.dev/docs/getting-started/cli)
+
+{% include toc.md %}
+
+## Add security with Auth0 and the Okta Spring Boot starter
```

Comment:
It's not clear to me at this point why we need an Auth0 account, OpenFGA is e standalone system.

Maybe add some context you use it to issue access tokens and get a user's ID

Reply:
Added the context. It is for authentication.